### ISSUE TITLE
[select] Use the `vaadin-select-list-box` and `vaadin-select-item` to align this with the web component

### ISSUE THREAD

**Issue by vursen** [vaadin/flow-components#2369]
Along with the `items` API introduced for the select web component, there have been also introduced the `<vaadin-select-list-box>` and `<vaadin-select-item>` internal components to be used instead of more generic `<vaadin-list-box>` and `<vaadin-item>` and make direct styling possible (https://github.com/vaadin/web-components/pull/2990).

However, the Flow Select counterpart still uses `<vaadin-list-box>` and `<vaadin-item>` which results in some integration tests failing.

The Flow counterpart should be aligned with the web component and use new components.

**Comment by yuriy-fix** [comment]
Change renderer to render `<vaadin-select-list-box>` and `<vaadin-select-item>`.
